core/src/main/java/com/novoda/merlin/MerlinsBeard.java

```diff
@@ -44,16 +48,34 @@ private NetworkInfo getNetworkInfo() {
     }
 
     /**
-     * Provides a boolean representing whether a wifi network connection has been established.
+     * Provides a boolean representing whether a Wi-Fi network connection has been established.
      *
      * NOTE: Therefore available does not necessarily mean that an internet connection
      * is available.
      *
-     * @return boolean true if a wifi network connection is available.
+     * @return boolean true if a Wi-Fi network connection is available.
      */
     public boolean isConnectedToWifi() {
         NetworkInfo networkInfo = connectivityManager.getNetworkInfo(ConnectivityManager.TYPE_WIFI);
         return networkInfo != null && networkInfo.isConnected();
     }
 
+
+    /**
+     * Provides a boolean representing whether a mobile network connection has been established.
+     * NOTE: Therefore available does not necessarily mean that an internet connection
+     * is available.
+     *
+     * @return boolean true if a mobile network connection is available.
+     */
+    public boolean isConnectedToMobileNetwork() {
+        int simState = telephonyManager.getSimState();
+        if (simState == TelephonyManager.SIM_STATE_READY) {
```

Comment:
Why don't we check for sim state in isConnected?
Am 29.06.2015 18:14 schrieb "Daniele" <notifications@github.com>:

> In core/src/main/java/com/novoda/merlin/MerlinsBeard.java
> <https://github.com/novoda/merlin/pull/59#discussion_r33481107>:
>
> >       */
> >      public boolean isConnectedToWifi() {
> >          NetworkInfo networkInfo = connectivityManager.getNetworkInfo(ConnectivityManager.TYPE_WIFI);
> >          return networkInfo != null && networkInfo.isConnected();
> >      }
> >
> > +
> > +    /**
> > +     * Provides a boolean representing whether a mobile network connection has been established.
> > +     * NOTE: Therefore available does not necessarily mean that an internet connection
> > +     * is available.
> > +     *
> > +     * @return boolean true if a mobile network connection is available.
> > +     */
> > +    public boolean isConnectedToMobileNetwork() {
> > +        int simState = telephonyManager.getSimState();
> > +        if (simState == TelephonyManager.SIM_STATE_READY) {
>
> Sorry, in that case I cannot understand your question.
> isConnected checks if there is an active connection, in general.
> isConnectedToMobileNetworks narrows the check for mobile networks only
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/novoda/merlin/pull/59/files#r33481107>.
>